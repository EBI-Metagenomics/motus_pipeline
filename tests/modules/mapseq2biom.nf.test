nextflow_process {

    name "Test module mapseq2biom"
    script "modules/mapseq2biom.nf"
    process "MAPSEQ2BIOM"

    test("mapseq2biom") {

        when {
            params {
                mapseq = "${baseDir}/tests/modules/fixtures/mapseq2biom/test-mapseq"
                otu_ref = "${baseDir}/tests/modules/fixtures/mapseq2biom/test-otu"
                otu_label = "test"
            }
            process {
                """
                input[0] = channel.fromPath("${params.mapseq}", checkIfExists: true)
                input[1] = channel.fromPath("${params.otu_ref}", checkIfExists: true)
                input[2] = channel.value("${params.otu_label}")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}