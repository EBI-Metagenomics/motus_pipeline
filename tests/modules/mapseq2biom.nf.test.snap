{
    "mapseq2biom": {
        "content": [
            {
                "mapseq2biom_txt": [
                    "test-mapseq.txt:md5,fec26a63bbdeaea136fd751640cc305f"
                ],
                "mapseq2biom_tsv": [
                    "test-mapseq.tsv:md5,41c4f5b5857339f066cd790dacbe4377"
                ],
                "mapseq2biom_notaxid": [
                    "test-mapseq.notaxid.tsv:md5,79924bc9316de373b87f073a8fdb291d"
                ]
            }
        ],
        "timestamp": "2023-02-28T17:20:32+0000"
    }
}